File: WebSite/EulerAList/pset32.cpp
#include <iostream>   
#include <algorithm>    
#include <cmath>
#include <string>
#include <vector>
#include <unordered_set>

using namespace std;
unordered_set<int> res;

int main () 
{
    long result = 0;
    int myints[] = {1,2,3,4,5,6,7,8,9};
    do 
    {
        long currValue = 0;
        for (int i = 0; i < 9; i ++)
        {
            currValue += myints[i] * pow(10, 8 - i);
        }
        for (int first = 0; first <= 4; first ++)
        {
            long x1 = currValue / (int)pow(10, 8 - first);
            long residual = currValue % (int)pow(10, 8 - first);
            for (int second = first + 1; second <= 5; second ++)
            {
                long x2 = residual / (int)pow(10, 8 - second);
                long y = residual % (int)pow(10, 8 - second);
                if (x1 * x2 == y)
                {
                    cout << x1 << "*" << x2 << "=" << y << endl;
                    res.insert(y);
                }
            }
        }


    } while (next_permutation(myints,myints+9));
    for (int i : res)
    {
        result += i;
    }
    cout << result << endl;
    return 0;
}

File: WebSite/EulerAList/pset32.py
from itertools import permutations

res = set()
result = 0
myints = [1,2,3,4,5,6,7,8,9]
for i in permutations(myints):
    currValue = 0
    for j in range(9):
        currValue += i[j] * pow(10, 8 - j)
    for first in range(5):
        x1 = currValue // int(pow(10, 8 - first))
        residual = currValue % int(pow(10, 8 - first))
        for second in range(first + 1, 6):
            x2 = residual // int(pow(10, 8 - second))
            y = residual % int(pow(10, 8 - second))
            if x1 * x2 == y:
                res.add(y)

for i in res:
    result += i
print(result)
